pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -61,10 +38,18 @@ const IncomingOrder = ({ user }) => {
         return
       }
 
-      // means a deliverer has already takent the order
+      // means a deliverer has already taken the order
       if (ordersData.deliverer_id) {
-        router.push('/home')
-        setAlert({ type: 'info', message: 'Order has been taken' })
+        const { error: removePendingError } = await supabaseClient.rpc(
+          'remove_pending_order',
+          { order_id: id, user_id: user.id },
+        )
+        if (removePendingError) {
+          setAlert({ type: 'warning', message: removePendingError.code + ': ' + removePendingError.message })
+        } else {
+          setAlert({ type: 'info', message: 'Order has been taken' })
+        }
+        router.push('/deliverer/home')
```

Comment:
we should select the order regardless of the deliverer_id here, so that we can differentiate between an actual error and an order being taken.

we should also remove from pending_orders array if a order has been taken (although i realise this should really have been done already when a deliverer accepts the order, will do in next commit)